### ISSUE TITLE
Yaml/toml/json builders.

### ISSUE THREAD

**Issue by dpc** [slog-rs/slog#26]
Implement a crate that reads a configuration file and creates a drain defined in it. `Box<Drain>` will be used all over the place, which is supported and perfectly fine.

**Comment by dpc** [comment]
Maybe write at least a proof of concept before freezing things in 1.0 ...

**Comment by dpc** [comment]
Will do after 1.0.

**Comment by dpc** [comment]
Prototyping...

**Comment by dpc** [comment]
https://github.com/slog-rs/config in progress...

**Comment by mehcode** [comment]
@dpc Perhaps we could collaborate here? I'm writing a general configuration library to be used by frameworks and libraries. 

The idea is a library (such as slog-rs) just uses it `config::get("output.stderr.type")` and an application or framework will define where the configuration comes from.

```
config::merge(config::File::with_name("Settings")).unwrap();
```

https://github.com/mehcode/config-rs

**Comment by dpc** [comment]
It is definitely something to consider. I will try to investigate more when I had some spare time.

**Comment by dpc** [comment]
@mehcode , I did some longer investigation.

* You could make first `Source` implementation a  `&str`/`String`. Then `File` can be built on top of it, and people can generally fetch `String` from anywhere they want. I wonder if it could even be `impl Source for std::fs::File`. Generally in `slog-config` I'd like to retain the ability to feed it `&str`, that's why I'm thinking about it.
* There are quite a few `features` used. Right now `slog-config` it's serde that requires `nightly` and that should be fixed when procedural macros land in stable (hopefully soon).
* I don't understand how sequences work. `slog-config` config file can have multiple `outpu` instances, each describing different logging destination.

**Comment by dpc** [comment]
Anyway, since I released `slog-config` 0.1, I'm closing this. It should be reasonably easy to add more stuff to it, and there was a PR already, so I think the task is done ATM.
